dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -808,13 +808,15 @@ public OpenFileHandle createFile(String path, CreateFilePOptions options)
       // Check if the target file already exists. If yes, return by throwing error.
       boolean overWrite = options.hasOverwrite() ? options.getOverwrite() : false;
       boolean exists = ufs.exists(path);
-      if (!overWrite && exists) {
-        throw new RuntimeException(
-            new FileAlreadyExistsException(
-                String.format("File %s already exists but no overwrite flag", path)));
-      } else if (overWrite) {
-        // client is going to overwrite this file. We need to invalidate the cached meta and data.
-        mMetaManager.removeFromMetaStore(path);
+      if (exists) {
+        if (overWrite) {
+          // client is going to overwrite this file. We need to invalidate the cached meta and data.
+          mMetaManager.removeFromMetaStore(path);
+        } else {
+          throw new RuntimeException(
+              new FileAlreadyExistsException(
+                  String.format("File %s already exists but no overwrite flag", path)));
+        }
       }
```

Comment:
what if `exists == False` and `overWrite == True`? In the old version, the code will do `mMetaManager.removeFromMetaStore(path);`, while in your version, it does nothing here.

Reply:
^

Reply:
i think we should do nothing in this case...? Or it is possible that a file could no exist in ufs but exist in MetaStore 🤔

Reply:
UFS may change, but Alluxio does not sync with UFS.
So, it is possible that `exists == False` but metadata still has a metadata cache.
We should invalidate this metadata.

Reply:
tyty